src/rust/bitbox02-rust/src/xpubcache.rs

```diff
@@ -127,24 +127,20 @@ mod tests {
         #[derive(Clone)]
         struct MockXpub(Vec<u32>);
 
-        static mut CHILD_DERIVATIONS: u32 = 0;
-        static mut ROOT_DERIVATIONS: u32 = 0;
+        static CHILD_DERIVATIONS: util::UnsafeSyncRefCell<u32> = util::UnsafeSyncRefCell::new(0);
+        static ROOT_DERIVATIONS: util::UnsafeSyncRefCell<u32> = util::UnsafeSyncRefCell::new(0);
 
         impl Xpub for MockXpub {
             fn derive(&self, keypath: &[u32]) -> Result<Self, ()> {
                 let mut kp = Vec::new();
                 kp.extend_from_slice(&self.0);
                 kp.extend_from_slice(keypath);
-                unsafe {
-                    CHILD_DERIVATIONS += keypath.len() as u32;
-                }
+                *CHILD_DERIVATIONS.borrow_mut() += keypath.len() as u32;
```

Comment:
It would be better if we used raw pointers and kept the `unsafe` blocks. Using this `UnsafeSyncRefCell` it is not obvious anymore where the unsafe code is.

Reply:
technically it is still unsafe / unsound, because when you did `unsafe impl Sync` you promised that you have appropriate synchronization primitives internally. RefCell only manages the borrowchecker at runtime instead of compile time.